### PR TITLE
Refactors tekton pipeline reconciler to use tektoninstallerset client

### PR DESCRIPTION
**Pull request by sm43**
similar to tekton triggers, removes duplicate code

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/common/common.go [hunk 1]
```diff
@@ -48,11 +48,7 @@ func PipelineReady(informer informer.TektonPipelineInformer) (*v1alpha1.TektonPi
 		}
 		return nil, err
 	}
-	upgradePending, err := CheckUpgradePending(ppln)
```
[the remark this answers is not in the document]
Comment (sm43) [reply pkg/reconciler/common/common.go:51]: we used to add a label to cr for notifying upgrade to TC
now we are avoding api calls in reconciler
so we update message in status and figure out depending that

File: pkg/reconciler/kubernetes/tektonpipeline/reconcile.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2019 The Tekton Authors
```
Comment (PuneetPunamiya) [inline pkg/reconciler/kubernetes/tektonpipeline/reconcile.go:2]: a small nit :D

File: pkg/reconciler/kubernetes/tektonpipeline/finalize.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+// FinalizeKind removes all resources after deletion of a TektonPipeline.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonPipeline) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	// Delete CRDs before deleting rest of resources so that any instance
+	// of CRDs which has finalizer set will get deleted before we remove
+	// the controller;s deployment for it
+	if err := r.manifest.Filter(mf.CRDs).Delete(); err != nil {
+		logger.Error("Failed to deleted CRDs for TektonPipeline")
```
Comment (savitaashture) [inline pkg/reconciler/kubernetes/tektonpipeline/finalize.go:39]:
```suggestion
		logger.Error("Failed to delete CRDs for TektonPipeline")
```

File: pkg/reconciler/kubernetes/tektonpipeline/finalize.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		return err
+	}
+
+	if err := r.installerSetClient.CleanupAll(ctx); err != nil {
+		logger.Error("failed to cleanup all installerset: ", err)
+		return err
+	}
+
+	if err := r.extension.Finalize(ctx, original); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
```
Comment (savitaashture) [inline pkg/reconciler/kubernetes/tektonpipeline/finalize.go:49]:
```suggestion
		logger.Error("Failed to finalize platform resources: ", err)
```

File: pkg/reconciler/kubernetes/tektonpipeline/metrics.go [hunk 5]
```diff
@@ -33,11 +34,6 @@ var (
 		stats.UnitDimensionless)
 )
 
-const (
-	metricsNew     = "NewInstall"
-	metricsUpgrade = "Upgrade"
-)
```
Comment (savitaashture) [inline pkg/reconciler/kubernetes/tektonpipeline/metrics.go:39]: How are we handling count of newInstall and upgrade ?

Comment (sm43) [reply pkg/reconciler/kubernetes/tektonpipeline/metrics.go:39]: we implement https://github.com/tektoncd/operator/blob/b32c133b7da53f614ddd28c26c8838a59ca1c0c4/pkg/reconciler/kubernetes/tektoninstallerset/client/metrics.go#L21
and the package handles emitting metrics for the component

**Comment by sm43** [comment]
@savitaashture updated!

**Review by savitaashture — COMMENTED** [review]
/lgtm

:tada:

**Pull request merged**